### DISCUSSION TITLE
Structured Logging in a stack that has Lambda and containerized services

### DISCUSSION THREAD

**Discussion by ssathasivanACR** [aws-powertools/powertools-lambda-dotnet#289] [General]
Hi All,

We have an architecture - REST APIs (implemented as API Gateway + Lambda) and worker processes (ECS /Fargate , SQS/Lambda, Event Bridge/SQS/Lambda).  

For the Lambda part f the stack, we would like to use .NET Power tool for logging. In order to implement structured logging in worker processes (ECS /Fargate), is it recommended to use .NET Power tools, so that log structure remains consistent across microservices.

Regards
Sabarish

**Comment by hjgraca** [comment]
Hi @ssathasivanACR ,

Powertools could be a good option, and you raise a good point on maintaining consistency across services running inside or outside Lambda. 

Cheers,
Henrique

**Comment by ssathasivanACR** [comment]
Hi @hjgraca ,

Is this something that is being discussed with in AWS

Regards
Sabarish

**Reply by hjgraca** [reply]
There was a discussion [here](https://github.com/awslabs/aws-lambda-powertools-python/discussions/1684).
In your example, using logging everything should work as expected.

**Reply by ssathasivanACR** [reply]
The discussion seems to be revolved around Tracing and Metrics. My question is around logging.

**Reply by hjgraca** [reply]
The discussion is around what could be less supported, logging is the most supported feature.
